tools/SchemaManager/Commands/ApplyCommand.cs

```diff
@@ -150,10 +150,18 @@ await Policy.Handle<SchemaManagerException>()
             }
         }
 
-        private static async Task UpgradeSchemaAsync(string connectionString, int version, string script, CancellationToken cancellationToken)
+        private static async Task UpgradeSchemaAsync(string connectionString, int version, string script, CancellationToken cancellationToken, bool isFullSchemaSnapshot = false)
         {
-            // check if the record for given version exists in failed status
-            await SchemaDataStore.DeleteSchemaVersionAsync(connectionString, version, SchemaDataStore.Failed, cancellationToken);
+            // Replace the check(version > 5) with SchemaInformation.DoesNotContainsVersionInSchema after Microsoft.Health.SqlServer dependency upgrade.
+            if (!isFullSchemaSnapshot || version > 5)
```

Comment:
>version > 5 [](start = 41, length = 11)

Don't you want to use `SchemaInformation.DoesNotContainsVersionInSchema(version)` here?